mailer.rst

```diff
@@ -0,0 +1,653 @@
+    ;
+
+Alternatively you can use the ``attach()`` method to attach contents from a stream::
+
+    $email = (new Email())
+        // ...
+        ->attach(fopen('/path/to/documents/contract.doc', 'r'))
+    ;
+
+Embedding Images
+~~~~~~~~~~~~~~~~
+
+If you want to display images inside your email, you must embed them
+instead of adding them as attachments. When using Twig to render the email
+contents, as explained :ref:`later in this article <embedding-images-in-emails-with-twig>`,
+the images are embedded automatically. Otherwise, you need to embed them manually.
+
+First, use the ``embed()`` or ``embedFromPath()`` method to add an image from a
+file or stream::
+
+    $email = (new Email())
+        // ...
+        // get the image contents from a PHP resource
+        ->embed(fopen('/path/to/images/logo.png', 'r'), 'logo')
+        // get the image contents from an existing file
+        ->embedFromPath('/path/to/images/signature.gif', 'footer-signature')
+    ;
+
+The second optional argument of both methods is the image name ("Content-ID" in
+the MIME standard). Its value is an arbitrary string used later to reference the
+images inside the HTML contents::
+
+    $email = (new Email())
+        // ...
+        ->embed(fopen('/path/to/images/logo.png', 'r'), 'logo')
+        ->embedFromPath('/path/to/images/signature.gif', 'footer-signature')
+        // reference images using the syntax 'cid:' + "image embed name"
+        ->html('<img src="cid:logo"/> ... <img src="cid:footer-signature"/> ...')
+    ;
+
+Global from Address
+-------------------
+
+Instead of calling ``->from()`` *every* time you create a new ``Email``, you can
+create an event subscriber to set it automatically::
+
+    // src/Mailer/MailerFromListener.php
+    namespace App\Mailer;
+
+    use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+    use Symfony\Component\Mailer\Event\MessageEvent;
+    use Symfony\Component\Mime\Email;
+
+    class MailerFromListener implements EventSubscriberInterface
+    {
+        public function onMessageSend(MessageEvent $event)
+        {
+            $message = $event->getMessage();
+
+            // make sure it's an Email object
+            if (!$message instanceof Email) {
+                return;
+            }
+
+            // always set the from address
+            $message->from('fabien@symfony.com');
+        }
+
+        public static function getSubscribedEvents()
+        {
+            return [MessageEvent::class => 'onMessageSend'];
+        }
+    }
+
+.. _mailer-twig-integration:
+
+Twig: HTML & CSS
+----------------
+
+The Mime component integrates with the :doc:`Twig template engine </templating>`
+to provide advanced features such as CSS style inlining and support for HTML/CSS
+frameworks to create complex HTML email messages. First, make sure Twig is instaled:
+
+.. code-block:: terminal
+
+    $ composer require twig
```

Comment:
```suggestion
    $ composer require symfony/twig-bundle
```

IIRC, we decided not to use aliases in the docs. (and you're not using it anywhere else in this guide)